cmd/icingadb/main.go

```diff
@@ -169,9 +171,25 @@ func run() int {
 	signal.Notify(sig, os.Interrupt, syscall.SIGTERM, syscall.SIGHUP)
 
 	go func() {
+		var callback func(database.Entity) bool
+		var callbackKeyStructPtr map[string]any
+
+		if cfg := cmd.Config.NotificationsSource; cfg.ApiBaseUrl != "" {
+			logger.Info("Starting Icinga Notifications source")
+
+			notificationsSource := notifications.NewNotificationsClient(
+				ctx,
+				db,
+				rc,
+				logs.GetChildLogger("notifications-source"),
+				cfg)
+			callback = notificationsSource.Submit
+			callbackKeyStructPtr = notifications.SyncKeyStructPtrs
+		}
```

Comment:
This shouldn't be done in this background routine given that `notifications.NewNotificationsClient()` can terminate the process:

https://github.com/Icinga/icingadb/blob/ed7b0d1e0cf739fde84f5f9aedb60f654061c85c/pkg/notifications/notifications.go#L65-L68

Also, look at the number of calls to `logger.Fatal*()` in this function. So for consistency, that function shouldn't even terminate the process by itself but leave the job to `main()`.